invokeai/app/api/sockets.py

```diff
@@ -13,24 +13,39 @@ class SocketIO:
 
     def __init__(self, app: FastAPI):
         self.__sio = SocketManager(app=app)
-        self.__sio.on("subscribe", handler=self._handle_sub)
-        self.__sio.on("unsubscribe", handler=self._handle_unsub)
 
+        self.__sio.on("subscribe_session", handler=self._handle_sub_session)
+        self.__sio.on("unsubscribe_session", handler=self._handle_unsub_session)
         local_handler.register(event_name=EventServiceBase.session_event, _func=self._handle_session_event)
 
+        self.__sio.on("subscribe_queue", handler=self._handle_sub_queue)
+        self.__sio.on("unsubscribe_queue", handler=self._handle_unsub_queue)
+        local_handler.register(event_name=EventServiceBase.queue_event, _func=self._handle_queue_event)
+
     async def _handle_session_event(self, event: Event):
         await self.__sio.emit(
             event=event[1]["event"],
             data=event[1]["data"],
             room=event[1]["data"]["graph_execution_state_id"],
         )
 
-    async def _handle_sub(self, sid, data, *args, **kwargs):
+    async def _handle_sub_session(self, sid, data, *args, **kwargs):
         if "session" in data:
             self.__sio.enter_room(sid, data["session"])
 
-        # @app.sio.on('unsubscribe')
-
-    async def _handle_unsub(self, sid, data, *args, **kwargs):
+    async def _handle_unsub_session(self, sid, data, *args, **kwargs):
         if "session" in data:
             self.__sio.leave_room(sid, data["session"])
+
+    async def _handle_queue_event(self, event: Event):
+        await self.__sio.emit(
+            event=event[1]["event"],
+            data=event[1]["data"],
+            room="queue",
+        )
+
+    async def _handle_sub_queue(self, sid, *args, **kwargs):
+        self.__sio.enter_room(sid, "queue")
```

Comment:
Why the constant `queue` name vs something dynamic? In a hosted environment, that would put everyone in the same room. I know thats not a concern here - and we don't use this version of the service - but curious what your thought process was and what benefits we get from this if any.

Reply:
The queue is a singleton entity so there's only the one to subscribe to. But also, I named it during the MVP phase, so maybe there's a better way to handle it

Reply:
I'm thinking we need to use queue_id or something so that every user isn't in the same room in a hosted environment.

Reply:
My plan was to have a route that returns the room id for queue events. On commercial, the route can return anything it wants, but on OSS, it will probably just return "queue" for simplicity. Does that sound ok?